numpy/_core/src/multiarray/nditer_pywrap.c

```diff
@@ -77,8 +76,14 @@ static int npyiter_cache_values(NewNpyArrayIterObject *self)
         self->innerloopsizeptr = NULL;
     }
 
-    /* The read/write settings */
-    NpyIter_GetReadFlags(iter, self->readflags);
+    if (self->writeflags == NULL) {
+        self->writeflags = PyMem_Malloc(sizeof(char) * NpyIter_GetNOp(iter));
```

Comment:
This should be deallocated somewhere, no? Presumably in `npyiter_dealloc`?

Reply:
ouch, yes thanks, this is a serious thing!